klasses/views.py

```diff
@@ -0,0 +1,76 @@
+"""
+Views for Klasses app
+"""
+import logging
+
+from django.contrib.auth.models import User
+from rest_framework import (
+    authentication,
+    permissions,
+    status,
+)
+from rest_framework.generics import get_object_or_404
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from backends.edxorg import EdxOrgOAuth2
+from klasses.api import get_user_klasses_info, get_user_klass_info
+from klasses.permissions import CanReadIfSelf
+
+log = logging.getLogger(__name__)
+
+
+class UserKlassList(APIView):
+    """
+    Class based view for user klasse list view.
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+    )
+    permission_classes = (
+        permissions.IsAuthenticated,
+        CanReadIfSelf,
+    )
+
+    def get(self, request, username, *args, **kargs):  # pylint: disable=unused-argument
+        """
+        Returns information about the payments for the user in all the klasses she is allowed to pay.
+        """
+        user = get_object_or_404(
+            User,
+            social_auth__uid=username,
+            social_auth__provider=EdxOrgOAuth2.name
+        )
+
+        return Response(get_user_klasses_info(user=user))
+
+
+class UserKlassDetail(APIView):
```

Comment:
i think you can use `GenericAPIView` ([doc](http://www.django-rest-framework.org/api-guide/generic-views/#genericapiview)) here to automatically fetch the klass associated with the `klass_id` passed into this view. `mail.views.LearnerMailView` in micromasters ([link](https://github.com/mitodl/micromasters/blob/master/mail/views.py#L46)) has an example of this usage. You would specify extra class-level properties, something like this:

```python
    lookup_field = "id"
    lookup_url_kwarg = "klass_id"
    queryset = Klass.objects.all()
```

The klass would then be available in the `get` method like this: `self.get_object()`